addons/mail/static/src/core/common/thread.scss

```diff
@@ -38,3 +38,9 @@
         display: none!important;
     }
 }
+
+.o-mail-Thread-inboxHint {
+    &:hover {
+        text-decoration: underline;
+    }
+}
```

Comment:
I would define a `o-hover-text-underline` utility class.
That way the classname is clearly used for the sole purpose of style, and it can be reused with ease.